Assets/Scripts/StaminaBarSetColour.cs

```diff
@@ -0,0 +1,98 @@
+
+public class StaminaBarSetColour : MonoBehaviour {
+    public SpriteRenderer spriteRenderer;
+
+    public Sprite sprite1;
+    public Sprite sprite2;
+    public Sprite sprite3;
+    public Sprite sprite4;
+    public Sprite sprite5;
+    public Sprite sprite6;
+    public Sprite sprite7;
+    public Sprite sprite8;
+    public Sprite sprite9;
+    public Sprite sprite10;
+    public Sprite sprite11;
+    public Sprite sprite12;
+    public Sprite sprite13;
+    public Sprite sprite14;
+    public Sprite sprite15;
+    public Sprite sprite16;
+    public Sprite sprite17;
+    public Sprite sprite18;
+    public Sprite sprite19;
+    public Sprite sprite20;
+    public Sprite sprite21;
+    public Sprite sprite22;
+    public Sprite sprite23;
+    public Sprite sprite24;
+    public Sprite sprite25;
+
+    public bool debug;
+
+	public void ChangeSprite(int stamina, int maxStamina) {
+        double value = getvalue(stamina, maxStamina);
+        if (debug)
+            Debug.Log("Value: " + value);
+
+        if (value == 0)
+            spriteRenderer.sprite = sprite25;
+        else if (value == 1)
+            spriteRenderer.sprite = sprite24;
+        else if (value == 2)
+            spriteRenderer.sprite = sprite23;
+        else if (value == 3)
+            spriteRenderer.sprite = sprite22;
+        else if (value == 4)
+            spriteRenderer.sprite = sprite21;
+        else if (value == 5)
+            spriteRenderer.sprite = sprite20;
+        else if (value == 6)
+            spriteRenderer.sprite = sprite19;
+        else if (value == 7)
+            spriteRenderer.sprite = sprite18;
+        else if (value == 8)
+            spriteRenderer.sprite = sprite17;
+        else if (value == 9)
+            spriteRenderer.sprite = sprite16;
+        else if (value == 10)
+            spriteRenderer.sprite = sprite15;
+        else if (value == 11)
+            spriteRenderer.sprite = sprite14;
+        else if (value == 12)
+            spriteRenderer.sprite = sprite13;
+        else if (value == 13)
+            spriteRenderer.sprite = sprite12;
+        else if (value == 14)
+            spriteRenderer.sprite = sprite11;
+        else if (value == 15)
+            spriteRenderer.sprite = sprite10;
+        else if (value == 16)
+            spriteRenderer.sprite = sprite9;
+        else if (value == 17)
+            spriteRenderer.sprite = sprite8;
+        else if (value == 18)
+            spriteRenderer.sprite = sprite7;
+        else if (value == 19)
+            spriteRenderer.sprite = sprite6;
+        else if (value == 20)
+            spriteRenderer.sprite = sprite5;
+        else if (value == 21)
+            spriteRenderer.sprite = sprite4;
+        else if (value == 22)
+            spriteRenderer.sprite = sprite3;
+        else if (value == 23)
+            spriteRenderer.sprite = sprite2;
+        else
+            spriteRenderer.sprite = sprite1;
+    }
+
+    private int getvalue(int stamina, int maxStamina)
```

Comment:
no not really, its getting a range from 1 to 24 based upon the ratio between the current and maximum stamina.

the reason this is 24 is because I used 25 sprites and call on that in a switch statement.